### ISSUE TITLE
Add the ability to capture/modify console output

### ISSUE THREAD

**Issue by jan64** [mc-server/MCServer#1802]
This could be a good thing for possible server management plugins

**Comment by worktycho** [comment]
This could be done by creating a new descendent of cLogger::cListener that wraps a lua callbacks plus a wrapper for cLogger::GetInstance and cLogger::AttachListener.

**Comment by jan64** [comment]
In that case there would be no way to change console output,
but it seems good enough for most of the purposes.

**Comment by worktycho** [comment]
Modifying output would involve creating filter infrastructure within cLogger. Could you give a use case for modifying the log?

**Comment by jan64** [comment]
The only legit use case i could think of would be writing log filter plugin. 
So, i guess it's not really that needed.

**Comment by worktycho** [comment]
I can't think of any use case where a log filter plugin would be useful that couldn't be equally well solved by better control of logging in the main server.

**Comment by worktycho** [comment]
Taking a quick survey of the bukkit plugins they have two main use cases, filtering logs from the vanilla server and fixing other plugins logs. For the first use case it is better for us to try and avoid emitting the log in the first place through config options. For the second use case it might be worthwhile but I think we should handle that when/if that becomes a problem.

**Comment by madmaxoft** [comment]
Just a quick note to consider: The logfile filter must be capable of being unloaded as well, for when the plugins are reloaded / unloaded.

**Comment by worktycho** [comment]
Yes. The wrapper would need to unregister itself on destruction.

**Comment by madmaxoft** [comment]
Maybe we should change how RCON works - instead of executing commands "separately" from the main console, the RCON client should receive all the console output while it is connected, and inject any incoming commands as if they came from the console.

**Comment by jan64** [comment]
That could break tools parsing returned command output.

**Comment by madmaxoft** [comment]
This is no longer an "easy" issue, Lua callbacks are at least moderately difficult, and handling unloading / reloading is quite difficult.

**Comment by madmaxoft** [comment]
With the introduction of `cLuaState::cCallback`, we are again one step closer to having this implemented.

**Comment by pwnorbitals** [comment]
Also pinging this one as it looks very old, have we had progress ? :)

**Comment by bearbin** [comment]
There's no need to "ping" old issues. If there's been nothing written, then there almost certainly hasn't been any progress. If you want to have a go at this, pllease go ahead :)

**Comment by pwnorbitals** [comment]
Well, my experience tells me that pinging issues when they get old usually leads to newer information, discussion and thoughts which wouldn't have been shared otherwise. Everyone helps the projects as well as they can :) Anyway, it's fine if there is no progress, it doesn't look very high-priority right now
